extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.tracking.js

```diff
@@ -0,0 +1,127 @@
+define('ext.wikia.design-system.on-site-notifications.tracking', [
+		'jquery',
+		'wikia.log',
+		'wikia.tracker',
+		'ext.wikia.design-system.on-site-notifications.common'
+	], function ($, log, tracker, common) {
+		'use strict';
+
+		function Tracking(model) {
+			this.model = model;
+			this.labels = {
+				'discussion-upvote-reply': 'discussion-upvote-reply',
+				'discussion-upvote-post': 'discussion-upvote-post',
+				'discussion-reply': 'discussion-reply',
+				'mark-all-as-read': 'mark-all-as-read',
+				'mark-as-read': 'mark-as-read',
+				'open-menu': 'open-menu'
+			};
+		}
+
+		Tracking.prototype = {
+
+
+			registerEventHandlers: function (view) {
+				view.onMarkAllAsReadClick.attach(this.markAllAsReadClick.bind(this));
+				view.onMarkAsReadClick.attach(function (_, id) {
+					this.markAsReadClick(id);
+				}.bind(this));
+				view.onNotificationRender.attach(function (_, id) {
+					this.notificationImpression(id);
+				}.bind(this));
+				view.onNotificationClick.attach(function (_, id) {
+					this.notificationClick(id);
+				}.bind(this));
+			},
+
+			notificationImpression: function (id) {
+				try {
+					log('impression - notification', log.levels.info, common.logTag);
+					var notification = this.findById(id);
+					this.trackImpression(this.toLabel(notification.type),
+						{value: this.booleanToGa(notification.isUnread)});
+				} catch (e) {
```

Comment:
I just don't want any possible issues with tracking to blow up in the user's face.